flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/SpecificCaseClassSerializer.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.scala.typeutils
+
+import java.io.ObjectInputStream
+import java.lang.invoke.{MethodHandle, MethodHandles}
+
+import org.apache.flink.api.common.typeutils.CompositeTypeSerializerUtil.delegateCompatibilityCheckToNewSnapshot
+import org.apache.flink.api.common.typeutils.TypeSerializerConfigSnapshot.SelfResolvingTypeSerializer
+import org.apache.flink.api.common.typeutils._
+import org.apache.flink.api.java.typeutils.runtime.TupleSerializerConfigSnapshot
+import org.apache.flink.api.scala.typeutils.SpecificCaseClassSerializer.lookupConstructor
+
+import scala.collection.JavaConverters._
+import scala.reflect.runtime.universe
+
+class SpecificCaseClassSerializer[T <: Product](
```

Comment:
Missing serial version UID.

Reply:
Missing Javadoc. Ideally it should explain why we need this class to replace the `CaseClassSerializer`.
`CaseClassSerializer` should also be deprecated.